src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/DataLayer.cs

```diff
@@ -0,0 +1,228 @@
+			var initializer = holder.Init(subItems, changes);
+
+			// Initializer contains a snapshot of items when the collection is added. It's used by parent to raise add for those items on its FlatView.
+			return (holder, initializer);
+		}
+
+		private DataLayer(DataLayer? parent, IBindableCollectionDataLayerStrategy layerStrategy, ISchedulerInfo? context)
+		{
+			_context = context;
+			_parent = parent;
+			_layerStrategy = layerStrategy;
+
+			(Items, View, _facets) = _layerStrategy.CreateView(this);
+		}
+
+		private DataLayerUpdate Init(IObservableCollection source, IUpdateContext context)
+		{
+			// Prepare the buffer responsible to manage the collection changes from both: collection updates and collection changes events
+			// Note: The tracker built here is for FUTURE CollectionChanged events. We MUST NOT use the 'initContext' for that.
+			var changesContext = _layerStrategy.CreateUpdateContext(VisitorType.CollectionChanged, TrackingMode.Auto);
+			var changesTracker = _layerStrategy.GetTracker(this, changesContext);
+			var changesBuffer = new DataLayerChangesBuffer(this, source, changesTracker);
+
+			// Then start the buffer in order to analyze the current items of the 'source'
+			// Here we use the context which cause this holder to be instanciated (i.e. the 'initContext')
+			var initTracker = _layerStrategy.GetTracker(this, context);
+			var initializer = changesBuffer.Initialize(initTracker);
+
+			_currentChangesBuffer = changesBuffer;
+
+			return initializer;
+		}
+
+		/// <summary>
+		/// Updates the root data layer
+		/// </summary>
+		public void Update(IObservableCollection source, TrackingMode mode)
+		{
+			if (_currentChangesBuffer is null)
+			{
+				throw new InvalidOperationException("Invalid state. You must invoke the Init first.");
+			}
+
+			if (object.ReferenceEquals(source, _currentChangesBuffer.Collection))
+			{
+				return;
+			}
+
+			var context = _layerStrategy.CreateUpdateContext(VisitorType.UpdateCollection, mode);
+			var tracker = _layerStrategy.GetTracker(this, context);
+			var initializer = PrepareUpdate(source, tracker);
+
+			// Schedule to flush the buffers to apply the update
+			// TODO uno : Here we can send the prevent the Schedule and allow a sync execution at a given time
+			//			 Like when the message reached the view
+
+			Schedule(initializer.Complete);
+		}
+
+		/// <summary>
+		/// Updates a child data layer
+		/// </summary>
+		internal DataLayerUpdate PrepareUpdate(IObservableCollection source, IUpdateContext context)
+		{
+			var tracker = _layerStrategy.GetTracker(this, context); // re-use the context of the parent layer
+			var update = PrepareUpdate(source, tracker);
+
+			return update;
+		}
+
+		private DataLayerUpdate PrepareUpdate(IObservableCollection source, ILayerTracker tracker)
+		{
+			var from = _nextChangesBuffer ?? _currentChangesBuffer ?? throw new InvalidOperationException("Invalid state. You must invoke the init first.");
```

Comment:
Init on what? Can the message be even more actionable?

Reply:
The `Init()` method the same object :)